python/lib/candidate.py

```diff
@@ -83,53 +84,95 @@ def create_candidate(self, db, participants_info):
             if 'age' in row:
                 self.age = row['age']
 
+            # three steps to find site:
+            #   1. try matching full name from 'site' column in participants.tsv in db
+            #   2. try extracting alias from pscid
+            #   3. try finding previous site in candidate table
+
             if 'site' in row:
+                # search site id in psc table by its full name
                 site_info = db.pselect(
                     "SELECT CenterID FROM psc WHERE Name = %s",
                     [row['site'], ]
                 )
                 if(len(site_info) > 0):
                     self.center_id = site_info[0]['CenterID']
-            else:
```

Comment:
I believe the logic inside this code block should work. Maybe it is skipped because the if part gets evaluated?

Reply:
Done, same logic as before. The same code is just below to match the third candidate site check.